### PR TITLE
feat: make the consistency level configurable.

### LINKED ISSUE #154: make consistency level configurable
set in the constructor

### PR DESCRIPTION
**Pull request by RahulMaganti47**
Fixes #154.

### CODE REVIEW

**Comment by djb15** [comment]
Why do we want to make this configurable when it's going to be an immutable anyway? Is there a world where we don't want finality for the WH endpoint?

**Comment by RahulMaganti47** [comment]
> Why do we want to make this configurable when it's going to be an immutable anyway? Is there a world where we don't want finality for the WH endpoint?

there may be integrators who'd like to choose their own consistency level (even if it is immutable - this allows them to make this choice upfront)

File: evm/test/Upgrades.t.sol [hunk 1]
```diff
@@ -41,6 +41,7 @@ contract TestUpgrades is Test, IManagerEvents, IRateLimiterEvents {
         0xcfb12303a19cde580bb4dd771639b0d26bc68353645571a8cff516ab2ee113a0;
     WormholeSimulator guardian;
     uint256 initialBlockTimestamp;
+    uint8 constant FAST_CONSISTENCY_LEVEL = 200;
```
Comment (djb15) [inline evm/test/Upgrades.t.sol:44]: Why are we using this consistency level in the tests? Intrigued more than anything

File: evm/src/WormholeEndpoint.sol [hunk 2]
```diff
@@ -15,7 +15,7 @@ abstract contract WormholeEndpoint is Endpoint, IWormholeEndpoint, IWormholeRece
 
     // TODO -- fix this after some testing
     uint256 public constant GAS_LIMIT = 500000;
-    uint8 public constant CONSISTENCY_LEVEL = 1;
+    uint8 public immutable consistencyLevel;
```
Comment (djb15) [inline evm/src/WormholeEndpoint.sol:18]: Can we add a docstring above this then to specify common consistency levels that integrators might want to use and why?

File: evm/test/Upgrades.t.sol (hunk 1 above)
Comment (RahulMaganti47) [reply evm/test/Upgrades.t.sol:44]: this is code for instant finality. for testnet @evan-gray mentioned this would better.

**Pull request closed without merging**